OPHD/States/PlanetSelectState.cpp

```diff
@@ -239,12 +239,14 @@ void PlanetSelectState::onMousePlanetEnter()
 
 	for (std::size_t i = 0; i < mPlanets.size(); ++i)
 	{
-		// FIXME: Ugly, will be difficult to maintain in the future.
 		if (mPlanets[i]->mouseHovering())
 		{
-			if (mPlanets[i]->attributes().type == Planet::PlanetType::Ganymede) { mPlanetDescription.text(constants::PLANET_DESCRIPTION_GANYMEDE); }
-			if (mPlanets[i]->attributes().type == Planet::PlanetType::Mars) { mPlanetDescription.text(constants::PLANET_DESCRIPTION_MARS); }
-			if (mPlanets[i]->attributes().type == Planet::PlanetType::Mercury) { mPlanetDescription.text(constants::PLANET_DESCRIPTION_MERCURY); }
+			auto iterator = constants::PLANET_DESCRIPTIONS.find(mPlanets[i]->attributes().type);
+
+			if (iterator != constants::PLANET_DESCRIPTIONS.end())
+			{
+				mPlanetDescription.text(iterator->second);
+			}
```

Comment:
You can potentially shorten this to:
```cpp
mPlanetDescription.text(constants::PLANET_DESCRIPTIONS.at(mPlanets[i]->attributes().type));
```

If the item does not exist in the `std::map`, the `at` method will `throw` and exception. Assuming you want an exception here of course.

Reply:
Ah right, I forgot about the `.at` function. I like the idea of adding an exception message.